packages/compiler/src/expression_parser/parser.ts

```diff
@@ -195,31 +202,35 @@ export class Parser {
     const stringSpans: {start: number, end: number}[] = [];
     const expressionSpans: {start: number, end: number}[] = [];
     let i = 0;
-    let inInterpolation = false;
+    let atInterpolation = false;
     let {start: interpStart, end: interpEnd} = interpolationConfig;
     while (i < input.length) {
-      if (!inInterpolation) {
+      if (!atInterpolation) {
         // parse until starting {{
         const start = i;
-        while (i < input.length && input.substring(i, i + interpStart.length) !== interpStart) {
+        while (i < input.length && !input.startsWith(interpStart, i)) {
           ++i;
         }
         const part = input.substring(start, i);
         strings.push(part);
         stringSpans.push({start, end: i});
 
-        inInterpolation = true;
+        atInterpolation = true;
       } else {
         // parse from starting {{ to ending }}
         const fullStart = i;
         const exprStart = fullStart + interpStart.length;
         let exprEnd = exprStart;
-        while (exprEnd < input.length &&
-               input.substring(exprEnd, exprEnd + interpEnd.length) !== interpEnd) {
+        while (exprEnd < input.length && !input.startsWith(interpEnd, exprEnd)) {
           ++exprEnd;
         }
         const fullEnd = exprEnd + interpEnd.length;
 
+        if (exprEnd >= input.length) {
+          i = fullStart;
+          break;
+        }
```

Comment:
```suggestion
        if (exprEnd >= input.length) {
          // We could not find the end of the interpolation. Exit before pushing to the expressions list.
          break;
        }
```
Added clarification comment and removed the reset of `i` which I don't think is needed.

Reply:
done